repos/system_upgrade/common/actors/commonleappdracutmodules/files/dracut/85sys-upgrade-redhat/do-upgrade.sh

```diff
@@ -268,6 +270,12 @@ do_upgrade() {
         rv=$?
     fi
 
+    if [ "$rv" -ne 0 ]; then
+        # set the upgrade flag to prevent the upgrade from running again when
+        # the emergency shell exits and the upgrade.target is restarted
+        "$NEWROOT/bin/touch" "$NEWROOT$LEAPP_FAILED_FLAG_FILE"
```

Comment:
ensure the directory exists:
```suggestion
        [ -d "$(dirname $NEWROOT$LEAPP_FAILED_FLAG_FILE)" ] || mkdir "$NEWROOT$LEAPP_FAILED_FLAG_FILE"
        "$NEWROOT/bin/touch" "$NEWROOT$LEAPP_FAILED_FLAG_FILE"
```